### PR TITLE
nginx auth_request: fix -skip-provider-button

### LINKED ISSUE #284: Removing login prompt
Is there any possibility to remove the webpage that prompts you to log in using the provider set in the configuration? It really has no purpose if you have only one provider, and it should redirect straight to the oauth provider's page.

Is there an option to do this, or have I missed it in the documentation?

### PR DESCRIPTION
**Pull request by abraithwaite**
fixes #284

### CODE REVIEW

File: oauthproxy_test.go [hunk 1]
```diff
@@ -409,20 +414,48 @@ func TestSignInPageIncludesTargetRedirect(t *testing.T) {
...
+	sip_test := NewSignInPageTest(true)
+	const endpoint = "/some/random/endpoint"
+
+	code, body := sip_test.GetEndpoint(endpoint)
+	assert.Equal(t, 302, code)
+
+	match := sip_test.sign_in_provider_regexp.FindStringSubmatch(body)
+	if match == nil {
+		t.Fatal("Did not find pattern in body: " +
+			signInRedirectPattern + "\nBody:\n" + body)
```
Comment (abraithwaite) [inline oauthproxy_test.go]: Fixed the wrong one.

**Comment by ploxiln** [comment]
Just to clarify: If in proxy mode, the `/sign_in` route should never be hit, because proxy does this:

```go
func (p *OAuthProxy) Proxy(rw http.ResponseWriter, req *http.Request) {
	status := p.Authenticate(rw, req)
	if status == http.StatusInternalServerError {
		p.ErrorPage(rw, http.StatusInternalServerError,
			"Internal Error", "Internal Error")
	} else if status == http.StatusForbidden {
		if p.SkipProviderButton {
			p.OAuthStart(rw, req)
		} else {
			p.SignInPage(rw, req, http.StatusForbidden)
		}
	} else {
		p.serveMux.ServeHTTP(rw, req)
	}
}
```

So the case that this fixes is if "nginx auth_request" mode is being used, and the user is setting the nginx `error_page` to `/oauth2/sign_in`. They can already get the behavior they want by setting the nginx `error_page` to `/oauth2/start` instead.

**Comment by abraithwaite** [comment]
Yep, doesn't really make a huge difference to me.  I'll try it by setting the `error_page` as it seems more clean.

Just thought it might make sense here just for consistency with the other path and for newbies like me who get confused when that setting doesn't work with the example nginx conf. :-)

**Comment by ploxiln** [comment]
yes probably still a worthwhile change

**Pull request merged**